src/System.Private.CoreLib/shared/System/Convert.Base64.Avx2.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+using System.Runtime.Intrinsics;
+using System.Runtime.Intrinsics.X86;
+
+namespace System
+{
+    public static partial class Convert
+    {
+        private static unsafe void EncodeBase64Avx(byte* input, int inputLength, byte* output, int outputLength)
+        {
+            // Based on "Base64 encoding with SIMD instructions" article by Wojciech Muła
+            // http://0x80.pl/notesen/2016-01-12-sse-base64-encoding.html
+            // Encode - https://github.com/WojciechMula/base64simd/blob/master/encode/encode.avx2.cpp
+            // Lookup - https://github.com/WojciechMula/base64simd/blob/master/encode/lookup.avx2.cpp (lookup_pshufb_improved)
+
+            byte* outputCurrent = output;
+            int stride = 2 * 4 * 3;
+
+            Vector256<byte> shiftLut = Vector256.Create(
+                0x47, 0xFC, 0xFC, 0xFC, 0xFC, 0xFC, 0xFC, 0xFC,
+                0xFC, 0xFC, 0xFC, 0xED, 0xF0, 0x41, 0x00, 0x00,
+                0x47, 0xFC, 0xFC, 0xFC, 0xFC, 0xFC, 0xFC, 0xFC,
+                0xFC, 0xFC, 0xFC, 0xED, 0xF0, 0x41, 0x00, 0x00);
+
+            Vector256<byte> asciiToStringMask = Vector256.Create(
+                0, 0x80, 1, 0x80, 2, 0x80, 3, 0x80,
+                4, 0x80, 5, 0x80, 6, 0x80, 7, 0x80,
+                0, 0x80, 1, 0x80, 2, 0x80, 3, 0x80,
+                4, 0x80, 5, 0x80, 6, 0x80, 7, 0x80);
+
+            Vector256<byte> shuffleMask = Vector256.Create((byte)
+                1, 0, 2, 1, 4, 3, 5, 4, 7, 6, 8, 7, 10, 9, 11, 10,
+                1, 0, 2, 1, 4, 3, 5, 4, 7, 6, 8, 7, 10, 9, 11, 10);
+
+            int i = 0;
+
+            for (; i < inputLength - stride - 1; i += stride)
+            {
+                Vector128<byte> lo = Sse2.LoadVector128(input + i);
+                Vector128<byte> hi = Sse2.LoadVector128(input + i + stride / 2);
+
+                Vector256<byte> @in = Avx2.Shuffle(Avx2.InsertVector128(lo.ToVector256(), hi, 1), shuffleMask);
+
+                Vector256<byte>   t0 = Avx2.And(@in, Vector256.Create(0x0fc0fc00).AsByte());
```

Comment:
You should hoist these AVX-constants before the loop, so they are kept in (ymm-) registers, instead of creating the in eacht iteration.

Reply:
I did it initially but got even worse output (spills?) will check again later

Reply:
> I did it initially but got even worse output (spills?)

Currently, `Create` is not yet optimized to cache constant vector by the compiler (I plan to make it in the next steps).
So, I suggest defining all the constant vector as `static` members.

`Create(T)` (one argument) may be an exception that can generate a single `broadcast` instruction on AVX2 CPUs.